cluster-autoscaler/utils/kubernetes/listers.go

```diff
@@ -153,67 +153,73 @@ type PodLister interface {
 	List() ([]*apiv1.Pod, error)
 }
 
-// UnschedulablePodLister lists unscheduled pods
-type UnschedulablePodLister struct {
+// ScheduledPodLister lists scheduled pods.
+type ScheduledPodLister struct {
 	podLister v1lister.PodLister
 }
 
-// List returns all unscheduled pods.
-func (unschedulablePodLister *UnschedulablePodLister) List() ([]*apiv1.Pod, error) {
-	var unschedulablePods []*apiv1.Pod
-	allPods, err := unschedulablePodLister.podLister.List(labels.Everything())
-	if err != nil {
-		return unschedulablePods, err
-	}
-	for _, pod := range allPods {
-		_, condition := podv1.GetPodCondition(&pod.Status, apiv1.PodScheduled)
-		if condition != nil && condition.Status == apiv1.ConditionFalse && condition.Reason == apiv1.PodReasonUnschedulable {
-			unschedulablePods = append(unschedulablePods, pod)
-		}
-	}
-	return unschedulablePods, nil
-}
-
-// NewUnschedulablePodLister returns a lister providing pods that failed to be scheduled.
-func NewUnschedulablePodLister(kubeClient client.Interface, stopchannel <-chan struct{}) PodLister {
-	return NewUnschedulablePodInNamespaceLister(kubeClient, apiv1.NamespaceAll, stopchannel)
+// List returns all scheduled pods.
+func (lister *ScheduledPodLister) List() ([]*apiv1.Pod, error) {
+	return lister.podLister.List(labels.Everything())
 }
 
-// NewUnschedulablePodInNamespaceLister returns a lister providing pods that failed to be scheduled in the given namespace.
-func NewUnschedulablePodInNamespaceLister(kubeClient client.Interface, namespace string, stopchannel <-chan struct{}) PodLister {
+// NewScheduledPodLister builds ScheduledPodLister
+func NewScheduledPodLister(kubeClient client.Interface, stopchannel <-chan struct{}) PodLister {
 	// watch unscheduled pods
-	selector := fields.ParseSelectorOrDie("spec.nodeName==" + "" + ",status.phase!=" +
+	selector := fields.ParseSelectorOrDie("spec.nodeName!=" + "" + ",status.phase!=" +
 		string(apiv1.PodSucceeded) + ",status.phase!=" + string(apiv1.PodFailed))
-	podListWatch := cache.NewListWatchFromClient(kubeClient.CoreV1().RESTClient(), "pods", namespace, selector)
+	podListWatch := cache.NewListWatchFromClient(kubeClient.CoreV1().RESTClient(), "pods", apiv1.NamespaceAll, selector)
 	store, reflector := cache.NewNamespaceKeyedIndexerAndReflector(podListWatch, &apiv1.Pod{}, time.Hour)
 	podLister := v1lister.NewPodLister(store)
 	go reflector.Run(stopchannel)
-	return &UnschedulablePodLister{
+
+	return &ScheduledPodLister{
 		podLister: podLister,
 	}
 }
 
-// ScheduledPodLister lists scheduled pods.
-type ScheduledPodLister struct {
+// ScheduledAndUnschedulablePodLister lists scheduled and unschedulable pods separately.
+type ScheduledAndUnschedulablePodLister interface {
+	List() ([]*apiv1.Pod, []*apiv1.Pod, error)
```

Comment:
Could you specify which return value is supposed to be which - here and in the List implementation below? For the interface, it's critical so that the implementations know which list to return in which value. For the implementation, it improves readability to always have the return values named if there are multiple values of the same type.

I'd also mention something about the two lists being obtained at the same point in time in the comment - the current wording actually sort of suggests something else (the "separately" part). Replacing "separately" with "obtained at the same point in time" should be enough.

Reply:
Done